src/plugins/intel_cpu/tests/functional/custom/behavior/ov_executable_network/properties.cpp

```diff
@@ -518,4 +520,63 @@ TEST_F(OVClassConfigTestCPU, smoke_CpuExecNetworkCheckCPURuntimOptionsWithCorePr
     ASSERT_EQ(valueCacheType.as<ov::element::Type>(), ov::element::bf16);
 }
 
+TEST_F(OVClassConfigTestCPU, smoke_CpuModelDistributionPolicyTensorParallel) {
+    ov::Core core;
+    std::shared_ptr<ov::Model> model = ov::test::utils::make_matmul_bias();
+    std::set<ov::hint::ModelDistributionPolicy> setModels = {ov::hint::ModelDistributionPolicy::TENSOR_PARALLEL};
+    ov::AnyMap config = {{ov::hint::model_distribution_policy.name(), setModels},
+                         {ov::num_streams.name(), 1},
+                         {ov::inference_num_threads.name(), 1}};
+
+    core.set_property(deviceName, config);
+    ov::CompiledModel compiledModel = core.compile_model(model, deviceName);
+
+    std::set<ov::hint::ModelDistributionPolicy> model_distribution_policy_value = {};
+    OV_ASSERT_NO_THROW(model_distribution_policy_value = compiledModel.get_property(ov::hint::model_distribution_policy));
+    ASSERT_EQ(model_distribution_policy_value, setModels);
+}
+
+TEST_F(OVClassConfigTestCPU, smoke_CpuModelDistributionPolicyTensorParallelAccurcay) {
```

Comment:
It has to be usual functional subgraph test, which already have all the infra for output comparision and inputs feeling. 
Besides that we have to cover cases like matmul with compressed weights.
It also worth to verify execution graph to make sure TP was actually applied

Reply:
Ok, I added test case about TP to `mat_mul.cpp` and `shared_matmul_weights_decompression.cpp`. Please have a look.